plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php

```diff
@@ -744,33 +745,29 @@ public function is_field( $key ) {
 	}
 
 	/**
-	 * Persists a field value for a given order. This would also optionally set the field value on the customer.
+	 * Persists a field value for a given order. This would also optionally set the field value on the customer object if the order is linked to a registered customer.
 	 *
-	 * @param string    $key The field key.
-	 * @param mixed     $value The field value.
-	 * @param \WC_Order $order The order to persist the field for.
-	 * @param bool      $set_customer Whether to set the field value on the customer or not.
+	 * @param string   $key The field key.
+	 * @param mixed    $value The field value.
+	 * @param WC_Order $order The order to persist the field for.
+	 * @param bool     $set_customer Whether to set the field value on the customer or not.
 	 *
 	 * @return void
 	 */
 	public function persist_field_for_order( $key, $value, $order, $set_customer = true ) {
 		$this->set_array_meta( $key, $value, $order );
-		if ( $set_customer ) {
-			if ( isset( wc()->customer ) ) {
-				$this->set_array_meta( $key, $value, wc()->customer );
-			} elseif ( $order->get_customer_id() ) {
-				$customer = new \WC_Customer( $order->get_customer_id() );
-				$this->set_array_meta( $key, $value, $customer );
-			}
+		if ( $set_customer && $order->get_customer_id() ) {
```

Comment:
No need to persist to session here; only for real customers.

Reply:
Why shouldn't we persist to session? If the same guest user was to make another order, they would need to fill the fields again?

Reply:
Ah okay I got it, this is only when persisting something to an order.